### DISCUSSION TITLE
Issuing betslips on Tari

### DISCUSSION THREAD

**Discussion by lessless** [tari-project/tari#5568] [General]
Hi peeps,

I'm building a small betting product in which I can  see a potential use case for Tari and hope someone can help me to  understanding  how Tari can fit into it.

Prior to making a bet user have to enter an email to which his betslip will be sent. Instead I would love to issue betslips on Tari network  and ideally replace emails with a digital assets issued on a Tari network. 

The flow goes like this: 

- after pressing "Pay" a modal dialog is shown to a user where he has to enter his email address
- after entering a valid email address and pressing "Next" user will see a LN invoice
- after paying an invoice a betslip  is sent to his email. Graphically betslip is a table that contains selected outcomes (13 rows).
- that betslip contains a link to page where user will be able to withdraw his prize
- to withdraw a prize user has to submit an LN invoice which will be paid in his favor
- that unique link in his email is the only authentication mechanism available. there is no concept of accounts, passwords, etc. 

Screencapture -> https://i.imgur.com/hJWWjFt.mp4

I would really love to start using Tari because it can enable an interesting use cases further down the road. Like trading bets or loyalty programs. 

Since I'm not yet deeply familiar with protocol I would love to get a better picture of how to implement this.

Particularly, I don't have capacity to build my own wallet app - **is it possible to manage betslips with Aurora? I.e. can I send betslips to Aurora address?** 

Here is what I have in my mind:
- after pressing "Pay" a modal dialog is shown to a user where he can enter his Aurora address
- user pays LN invoice and then a betslip is sent to his Aurora address. Ideally that betslip will contain a link to a withdrawal page.
- if user won he can go to a withdrawal page, submit an LN invoice and get his prize. 

This also raises few questions:
- how do I present a betslip with Tari? There is a requirement to  for it to contain selected outcomes -  a table with 13 rows. I guess that would be too much for a digital assset and  instead  information can be shown on a dedicated page (as an option). 
- is it possible to embed a link to betslip details page into a digital asset representing betslip? Additionally,  on that page user will be able to withdraw his prize. 
-  what authentication mechanisms can I use apart from unique link in a betslip? A wallet address? 

As you see there is a lot of confusion and uncertainity and I hope someone will help me clarify all those things. 

Thanks, you are awesome 😊!

**Comment by lessless** [comment]
Btw, here is a draft of a betslip email https://mjml.io/try-it-live/Hk596UFi8, just to give an idea how it looks like.

It contains a table with selected outcomes in 13 matches.

**Comment by CjS77** [comment]
The good news is that we want to make it easy for guys like you to build digital asset applications on top of Tari.

The bad news is that we're some distance away from having that functionality ready. Right now, we're stabilising the base layer (mimblewimble token); and working on some frameworks for digital assets (see the [validator node](https://github.com/tari-project/validator-node) repo).

**Comment by lessless** [comment]
Thanks @CjS77. It would be ideal to use Tari address as an authentication mechanis in a same way as email is used.
Then it woill be able to isse promotional credits for the first time user, for example: if user provides a Tari address before paying an invoice, then bet cost can be subtracted from his promotional credits account. 

I'm 100% sure Tari will change digital landscape, and all excited to be its early adopter. Godspeed folks!

**Comment by CjS77** [comment]
Indeed!
